### PR TITLE
XD75 keymap

### PR DESCRIPTION
**Pull request by ArpinFidel**
TODO :
- add option for backlight brightness 
- add dynamic macros

### CODE REVIEW

File: keyboards/xd75/keymaps/ArpinFidel/config.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/xd75/keymaps/ArpinFidel/config.h]: Could you replace this block: 
```c
#ifndef CONFIG_USER_H
#define CONFIG_USER_H

#include "../../config.h"
```
with just 
```h
#pragma once
```

And remove the `#endif` at the end?

File: keyboards/xd75/keymaps/ArpinFidel/keymap.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "xd75.h"
```
Comment (drashna) [inline keyboards/xd75/keymaps/ArpinFidel/keymap.c]: Please replace the include here with `#include QMK_KEYBOARD_H`

File: keyboards/xd75/keymaps/ArpinFidel/rules.mk [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/xd75/keymaps/ArpinFidel/rules.mk]: If you could remove the ifdef and include lines here?  They're a hold over from the old makefile system and should be removed

**Comment by ArpinFidel** [comment]
Sure. Just a sec

File: keyboards/xd75/keymaps/arpinfidel/config.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef CONFIG_USER_H
```
Comment (drashna) [inline keyboards/xd75/keymaps/arpinfidel/config.h]: Could you replace this block: 
```c
#ifndef CONFIG_USER_H
#define CONFIG_USER_H

#include "../../config.h"
```
with just 
```h
#pragma once
```

And remove the `#endif` at the end?

File: keyboards/xd75/keymaps/arpinfidel/keymap.c [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "xd75.h"
```
Comment (drashna) [inline keyboards/xd75/keymaps/arpinfidel/keymap.c]: Please replace the include here with `#include QMK_KEYBOARD_H`

File: keyboards/xd75/keymaps/arpinfidel/rules.mk [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ifndef QUANTUM_DIR
```
Comment (drashna) [inline keyboards/xd75/keymaps/arpinfidel/rules.mk]: If you could remove the ifdef and include lines here?  They're a hold over from the old makefile system and should be removed.

Basically you should have just:
```makefile
BOOTMAGIC_ENABLE = no
BACKLIGHT_ENABLE = yes
```

File: keyboards/xd75/keymaps/arpinfidel/config.h (hunk 4 above)
Comment (ArpinFidel) [reply keyboards/xd75/keymaps/arpinfidel/config.h]: should I do this with the default keymap as well? I copied those from the default

**Comment by drashna** [comment]
Awesome, thanks!

**Pull request merged**